### PR TITLE
Improves snapshot logging and snapshot deletion error handling

### PR DESCRIPTION
**Pull request by abeyad**
This commit does two things:
  1. Adds logging at the DEBUG level for when the index-N blob is
  updated.
  2. When attempting to delete a snapshot, if the snapshot was not found
  in the repository data, an exception is now thrown instead of silently
  ignoring the lack of presence of the snapshot in the repository data.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/repositories/RepositoryData.java [hunk 1]
```diff
@@ -189,8 +190,11 @@ public RepositoryData addSnapshot(final SnapshotId snapshotId,
      */
     public RepositoryData removeSnapshot(final SnapshotId snapshotId) {
         Map<String, SnapshotId> newSnapshotIds = snapshotIds.values().stream()
-            .filter(id -> snapshotId.equals(id) == false)
+            .filter(id -> !snapshotId.equals(id))
             .collect(Collectors.toMap(SnapshotId::getUUID, Function.identity()));
+        if (newSnapshotIds.size() == snapshotIds.size()) {
+            throw new ElasticsearchException("Attempting to remove non-existent snapshot [{}] from repository data", snapshotId);
```
Comment (imotov) [inline core/src/main/java/org/elasticsearch/repositories/RepositoryData.java]: Perhaps, something like `ResourceNotFoundException` would be more appropriate here?

**Review by imotov — APPROVED** [review]
Left a small suggestion. Otherwise LGTM.

File: core/src/main/java/org/elasticsearch/repositories/RepositoryData.java (hunk 1 above)
Comment (abeyad) [inline core/src/main/java/org/elasticsearch/repositories/RepositoryData.java]: ++, i made the change

**Comment by abeyad** [comment]
thanks for the review @imotov

**Comment by abeyad** [comment]
5.x commit: 309306b294ce6b61e5cd94bf2152bf52299826a6
5.5 commit: 662008c0ef9ebf89aee2d02cd5a1feb9d3bd98b0
5.4 commit: 52825e1ceaaa881e8a980c8a735da0c2be21782e

**Pull request merged**